apps/web/src/components/ProjectSwitcher.tsx

```diff
@@ -0,0 +1,170 @@
+"use client";
+
+import * as React from "react";
+import {
+  CaretSortIcon,
+  CheckIcon,
+  PlusCircledIcon,
+} from "@radix-ui/react-icons";
+
+import { cn } from "lib/utils";
+import { Avatar, AvatarFallback, AvatarImage } from "components/ui/avatar";
+import { Button } from "components/ui/button";
+import {
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+  CommandSeparator,
+} from "components/ui/command";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "components/ui/dialog";
+import { Input } from "components/ui/input";
+import { Label } from "components/ui/label";
+import { Popover, PopoverContent, PopoverTrigger } from "components/ui/popover";
+import { useProjectId } from "lib/hooks/useProjectId";
+import { trpc } from "utils/trpc";
+import { useSession } from "next-auth/react";
+import { useRouter } from "next/router";
+
+type PopoverTriggerProps = React.ComponentPropsWithoutRef<
+  typeof PopoverTrigger
+>;
+
+interface ProjectSwitcherProps extends PopoverTriggerProps {}
+
+export default function ProjectSwitcher({ className }: ProjectSwitcherProps) {
+  const currentProjectId = useProjectId();
+  const { data: projects, isLoading } = trpc.user.getProjects.useQuery();
+
+  const currentProject = projects?.projects.find(
+    ({ project }) => project.id === currentProjectId
+  );
+
+  const { update: sessionUpdate } = useSession();
+  const router = useRouter();
+  const [open, setOpen] = React.useState(false);
+  const [showNewTeamDialog, setShowNewTeamDialog] = React.useState(false);
+
+  return (
+    <Dialog open={showNewTeamDialog} onOpenChange={setShowNewTeamDialog}>
+      <Popover open={open} onOpenChange={setOpen}>
+        <PopoverTrigger asChild>
+          <Button
+            variant="outline"
+            role="combobox"
+            aria-expanded={open}
+            aria-label="Select a team"
+            className={cn("w-[200px] justify-between", className)}
+          >
+            <Avatar className="mr-2 h-5 w-5">
+              <AvatarImage
+                src={`https://avatar.vercel.sh/${currentProject?.project.id}.png`}
```

Comment:
Would be cool if a user could change the icon for a project, so that they could use their companies logo. In case they have many projects, it could make it easier to switch to the correct one, without reading all the project titles. Could be a seperate Issue though.

Reply:
Agreed, but should not be part of this PR imo #90